frontend/src/components/common/form/FastDatePicker.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FunctionComponent, memo, useEffect } from 'react';
+import { ErrorMessage, getIn, FormikProps } from 'formik';
+import DatePicker, { ReactDatePickerProps } from 'react-datepicker';
+import moment from 'moment';
+import { FormGroup, FormControlProps } from 'react-bootstrap';
+import { formikFieldMemo } from 'utils';
+
+type RequiredAttributes = {
+  /** The field name */
+  field: string;
+  /** formik state used for context and memo calculations */
+  formikProps: FormikProps<any>;
+};
+
+type OptionalAttributes = {
+  /** Specifies that the HTML element should be disabled */
+  disabled?: boolean;
+  /** Class name of the input wrapper */
+  outerClassName?: string;
+};
+
+export type FastDatePickerProps = FormControlProps &
+  RequiredAttributes &
+  OptionalAttributes &
+  Partial<ReactDatePickerProps>;
+
+/**
+ * Formik connected react-datepicker. Uses memo and cleanup inspired by
+ * https://jaredpalmer.com/formik/docs/api/fastfield
+ */
+const FormikDatePicker: FunctionComponent<FastDatePickerProps> = ({
+  field,
+  disabled,
+  outerClassName,
+  formikProps: {
+    values,
+    errors,
+    touched,
+    setFieldValue,
+    registerField,
+    unregisterField,
+    handleBlur,
+  },
+  ...rest
+}) => {
+  const error = getIn(errors, field);
+  const touch = getIn(touched, field);
+  let value = getIn(values, field);
+  if (value === '') {
+    value = null;
+  }
+  if (typeof value === 'string') {
+    value = moment(value, 'YYYY-MM-DD').toDate();
+  }
+  useEffect(() => {
+    registerField(field, {});
+    return () => {
+      unregisterField(field);
+    };
+  }, []);
+  const isInvalid = error && touch ? 'is-invalid ' : '';
+  const isValid = !error && touch && value && disabled ? 'is-valid ' : '';
+  return (
+    <FormGroup className={outerClassName ?? ''}>
+      <span className={isInvalid}></span>
+      <DatePicker
+        name={field}
+        value={value ?? moment()}
+        placeholderText="Enter date"
+        className={['form-control', 'date-picker', isInvalid, isValid].join(' ')}
+        dateFormat="MM/dd/yyyy"
+        selected={(value && new Date(value)) || null}
+        onBlur={handleBlur}
+        disabled={disabled}
+        {...rest}
+        onChange={(val: any) => {
+          setFieldValue(
+            field,
+            val
+              ? moment(val)
+                  .add('day', 1)
```

Comment:
Could you please explain why we need to add an extra day here?

Also, please note that the `add` syntax above is [deprecated](https://momentjs.com/docs/#/manipulating/add/) - could you please use the current one? thanks
```js
moment().add('seconds', 1); // Deprecated in 2.8.0
moment().add(1, 'seconds');
```

Reply:
I'll look into this, this may have been some code added during troubleshooting.